### PR TITLE
Add scripts to setup a physical machine to simulate a Gimlet

### PR DESCRIPTION
**Pull request by bnaecker**
- Add the `./tools/{create,destroy}_virtual_hardware.sh` scripts. These
  add some network interfaces and ZFS management commands which are
  required to run the real sled agent implementation. The networking
  bits are required to integrate OPTE and the xde kernel module.
- Adds `./tools/install_opte.sh` which downloads and installs the XDE
  kernel driver and corresponding kernel bits. It's called on illumos
  machines during `./tools/install_prerequisites.sh`
- Updates `docs/how-to-run.adoc` with new details

### CODE REVIEW

File: docs/how-to-run.adoc [hunk 1]
```diff
@@ -112,3 +112,19 @@ unique local addresses in `fd00:1de::/16`:
...
+`./tools/create_virtual_hardware.sh` script can be used. This creates a few
+file-based ZFS vdevs and ZFS zpools on top of those, and a couple of VNICs. The
+vdevs model the actual U.2s that will be in a Gimlet, and the VNICs model the
+two Chelsio NIC ports.
+
+You can clean up these resources with `./tools/destroy_virtual_hardware.sh`. Be
+aware that this is a best effort script. If the sled agent or other Omicron
+zones are still running, the zpools cannot be deleted. The script will warn you
+about the things it cannot remove, so you can do so yourself. Also note that all
+the networking bits are temporary, so a reboot should always clear them.
```
Comment (smklein) [inline docs/how-to-run.adoc:130]: Can we move this up above, before "Deploying Omicron"? I want this README to make sense from top-to-bottom - developers should consider running this before "install"

File: tools/create_virtual_hardware.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#
+# Make me a Gimlet!
+#
+# The entire control plane stack is designed to run and operate on the Oxide
+# rack, in each Gimlet. But Gimlet's don't quite exist yet. In the meantime,
+# this script can be used to create a few pieces of virtual hardware that
+# _simulate_ a Gimlet, allowing us to develop software that approximates the
+# eventual operation on Oxide hardware.
+#
+# See `docs/virtual-hardware.adoc` for more details.
```
Comment (smklein) [inline tools/create_virtual_hardware.sh:11]: This doc doesn't exist; `how-to-run.adoc` instead?

File: tools/install_prerequisites.sh [hunk 3]
```diff
@@ -100,6 +100,15 @@ fi
 ./tools/ci_download_cockroachdb
 ./tools/ci_download_clickhouse
 
+# Install OPTE
+#
+# OPTE is a Rust package that is consumed by a kernel module called xde. This
+# installs the `xde` driver and some kernel bits required to work with that
+# driver.
+if [[ "${HOST_OS}" == "SunOS" ]]; then
+    ./tools/install_opte.sh
```
Comment (smklein) [inline tools/install_prerequisites.sh:109]: Should we `pfexec` this?

File: tools/install_opte.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    local FILENAME="$(basename "$URL")"
+    local OUT_PATH="$OUT_DIR/$FILENAME"
+    local SHA="$2"
+    curl -L -o "$OUT_PATH" "$URL" 2> /dev/null
+    local ACTUAL_SHA="$(sha256sum "$OUT_PATH" | cut -d ' ' -f 1)"
+    if [[ "$ACTUAL_SHA" != "$SHA" ]]; then
+        echo "SHA mismatch downloding file $FILENAME"
+        exit 1
+    fi
+}
```
Comment (smklein) [inline tools/install_opte.sh:36]: Can we do a conditional download here? If it already exists and the SHA is valid, can we use the local file?

(This is what we do for CRDB / clickhouse now)

File: docs/how-to-run.adoc (hunk 1 above)
Comment (smklein) [reply docs/how-to-run.adoc:130]: Also, can we recommend running these with `pfexec`?

Comment (bnaecker) [reply docs/how-to-run.adoc:130]: Yes to both.

File: tools/install_prerequisites.sh (hunk 3 above)
Comment (bnaecker) [reply tools/install_prerequisites.sh:109]: I'd personally rather require users to run the script with `pfexec` itself, but I don't feel strongly.

File: tools/install_opte.sh (hunk 4 above)
Comment (bnaecker) [reply tools/install_opte.sh:36]: We can, yes. Will update.

File: tools/create_virtual_hardware.sh (hunk 2 above)
Comment (bnaecker) [reply tools/create_virtual_hardware.sh:11]: Thanks, meant to add some details about why this is all needed in that file. Will add.

File: tools/install_prerequisites.sh (hunk 3 above)
Comment (smklein) [reply tools/install_prerequisites.sh:109]: but also, that was a weak preference. If we want to mandate that the entire `install_prerequisites.sh` script be executed with sudo/pfexec, we can do that too

Comment (bnaecker) [reply tools/install_prerequisites.sh:109]: I see a `pfexec` in `install_prerequisites.sh:69`, installing the packages?

Comment (bnaecker) [reply tools/install_prerequisites.sh:109]: I also don't care too much, I had expected it'd be better for people to have to type `pfexec` to know that the script is...mucking about...but it should be pretty obvious 🤷

Comment (bnaecker) [reply tools/install_prerequisites.sh]: Done in c217a4fb

**Review by smklein — APPROVED** [review]

**Pull request merged**